v2/internal/testcommon/vcr/v3/test_recorder.go

```diff
@@ -92,7 +92,7 @@ func NewTestRecorder(
 		}
 
 		// verify custom request count header (see counting_roundtripper.go)
-		if r.Header.Get(vcr.COUNT_HEADER) != i.Headers.Get(vcr.COUNT_HEADER) {
+		if r.Header.Get(COUNT_HEADER) != i.Headers.Get(COUNT_HEADER) {
```

Comment:
No need to check body hash here because we're actually checking the body matches exactly, is that right?

Is it worth leaving a comment to that effect?

Reply:
Good catch. With the hash header, we don't need to read the entire body to do the match.